File: ADS 2024.1/POO/atividades_enviar/exercicio3/q02.java
public static boolean ehPrimo(int num) {
    if (num <= 1) {
        return false;
    }

    for (int i = 2; i <= Math.sqrt(num); i++) {
        if (num % i == 0) {
            return false;
        }
    }

    return true;
}

System.out.println(ehPrimo(7)); // true
System.out.println(ehPrimo(12)); // false

File: ADS 2024.1/POO/atividades_enviar/exercicio3/q02.ts
function ehPrimo(num: number): boolean {
    if (num <= 1) {
      return false;
    }
  
    for (let i = 2; i <= Math.sqrt(num); i++) {
      if (num % i === 0) {
        return false;
      }
    }
  
    return true;
}

console.log(ehPrimo(7)); // true
console.log(ehPrimo(12)); // false